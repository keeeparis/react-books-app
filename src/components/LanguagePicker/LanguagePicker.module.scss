.Wrapper {
  display: flex;
  gap: 16px;

  button span {
    font-size: 1.5em;
    color: black;
  }

  > div {
    width: 100px;
    color: black;
  }
}

@media (max-width: 660px) {
  .Wrapper > div {
    width: 85px;
    color: white;
    background-color: transparent;
    box-shadow: none;
    margin-right: -10px;

    &::after {
      border-color: inherit;
    }

    > div,
    > div:hover {
      background-color: transparent;
    }
  }
}
