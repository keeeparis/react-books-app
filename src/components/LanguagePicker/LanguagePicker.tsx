import React, { useMemo } from 'react'
import { useTranslation } from 'react-i18next'

import Select from '../Select'
import styles from './LanguagePicker.module.scss'

const LanguagePicker = () => {
  const { i18n } = useTranslation()

  const languages = useMemo(() => [...i18n.languages], [])

  const changeLanguage = (lang: string) => {
    i18n.changeLanguage(lang)
  }

  const transformInitialLanguage = (lang: string) => lang.split('-')[0]

  return (
    <div className={styles.Wrapper}>
      <Select
        data={languages}
        onClick={changeLanguage}
        value={transformInitialLanguage(i18n.language)}
      />
    </div>
  )
}

export default LanguagePicker
