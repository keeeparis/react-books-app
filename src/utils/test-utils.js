import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import 'regenerator-runtime/runtime'
import { Provider } from 'react-redux'
import { store } from '../redux/store/store'
import { MemoryRouter } from 'react-router-dom'
import '../i18n/config'

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  )
}

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { userEvent }
export * from 'regenerator-runtime/runtime'
export * from '../i18n/config'

export { customRender as render }
