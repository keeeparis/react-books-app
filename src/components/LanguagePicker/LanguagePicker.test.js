import { render, screen, fireEvent, userEvent } from 'test-utils'
import LanguagePicker from './LanguagePicker'

it('test LanguagePicker Component', () => {
  const { asFragment } = render(<LanguagePicker />)

  expect(asFragment()).toMatchSnapshot()
})
