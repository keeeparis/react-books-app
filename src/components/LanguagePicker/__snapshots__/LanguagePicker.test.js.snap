// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LanguagePicker Component 1`] = `
<DocumentFragment>
  <div
    class="Wrapper"
  >
    <div
      class="select"
    >
      <div
        class="selectShow"
        role="selected"
        tabindex="0"
      >
        en
      </div>
    </div>
  </div>
</DocumentFragment>
`;
